### PR TITLE
feat: add option for binary formatting

### PR DESCRIPTION
**Pull request by alangsto**
Allow the option to specify the encoding for binary formatting, instead of always defaulting to hex.

### CODE REVIEW

File: tests/test_snowflake.py [hunk 1]
```diff
@@ -327,37 +327,7 @@ def test_export_snowflake_table_to_s3_overwrite(mock_sf_connection):  # noqa: F8
...
         mock_cursor.execute.assert_has_calls(
             [
-                mock.call("\n        COPY INTO 's3://edx-test/test/test_database-test_schema-test_table/'\n            FROM test_database.test_schema.test_table\n            STORAGE_INTEGRATION = test_storage_integration\n            FILE_FORMAT = ( TYPE = CSV EMPTY_FIELD_AS_NULL = FALSE\n            FIELD_DELIMITER = ',' FIELD_OPTIONALLY_ENCLOSED_BY = NONE\n            ESCAPE_UNENCLOSED_FIELD = '\\\\'\n            NULL_IF = ( 'NULL' )\n            COMPRESSION = NONE\n            )\n            OVERWRITE=True\n            SINGLE=False\n            DETAILED_OUTPUT = TRUE\n            MAX_FILE_SIZE = 104857600\n    "),  # noqa
-            ]
-        )
-
-        mock_delete_s3_directory.assert_called_once_with('edx-test', 'test/test_database-test_schema-test_table/')
-
-
-def test_export_snowflake_table_to_s3_no_escape(mock_sf_connection):  # noqa: F811
```
Comment (alangsto) [inline tests/test_snowflake.py:337]: Not sure why this delete is being formatted like this, but there were two tests with the same name. I removed this one (it's the same test as the above test, but the same name as the below test).

**Review by ashultz0 — APPROVED** [review]

**Review by ashultz0 — CHANGES_REQUESTED** [review]
actually I take it back, I don't think this does anything despite looking like it

from the snowflake docs it won't do anything on export

BINARY_FORMAT = HEX | BASE64 | UTF8
Use
Data loading only

**Review by ashultz0 — APPROVED** [review]
ugh, now I don't know because I found "loading only" on this field when talking about file format
https://docs.snowflake.com/en/sql-reference/sql/alter-file-format.html
but https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html#format-type-options-formattypeoptions implies that it works on output so I guess safe to give it a try but maybe it will do nothing

**Comment by alangsto** [comment]
@ashultz0 I was reading https://docs.snowflake.com/en/sql-reference/sql/copy-into-location.html and https://community.snowflake.com/s/question/0D50Z00007b48V0SAI/encoding-format-of-file-while-exporting-larger-dataset-from-snowflake-to-external-storage-like-s3-or-azure when trying to decide next steps for debugging.

**Pull request merged**